scvi/inference/annotation.py

```diff
@@ -100,17 +102,22 @@ class ClassifierTrainer(Trainer):
         >>> trainer.test_set.accuracy()
     """
 
-    def __init__(self, *args, train_size=0.8, sampling_model=None, sampling_zl=False, use_cuda=True, **kwargs):
+    def __init__(self, *args, train_size=0.8, test_size=None, sampling_model=None, sampling_zl=False, use_cuda=True, **kwargs):
         self.sampling_model = sampling_model
         self.sampling_zl = sampling_zl
         super().__init__(*args, use_cuda=use_cuda, **kwargs)
-        self.train_set, self.test_set = self.train_test(self.model, self.gene_dataset,
-                                                        train_size=train_size,
-                                                        type_class=AnnotationPosterior)
+        self.train_set, self.test_set, self.validate_set = self.train_test_validate(self.model, self.gene_dataset,
+                                                                                    train_size=train_size,
+                                                                                    test_size=test_size,
+                                                                                    type_class=AnnotationPosterior)
         self.train_set.to_monitor = ['accuracy']
-        self.test_set.to_monitor = ['accuracy']
+        if self.test_set.nb_cells > 0:
```

Comment:
This lines are useless/wrong as even if `self.test_set.nb_cells = 0`, the metrics will still fail. 
I did a recent PR to prevent calculation of metrics at `compute_metrics` time.
So I think you can remove the if statement.